test/problems.jl

```diff
@@ -499,7 +499,7 @@ function climacore_2Dheat_test_cts(::Type{FT}) where {FT}
     jacobian = ClimaCore.MatrixFields.FieldMatrix((@name(u), @name(u)) => FT(-1) * LinearAlgebra.I)
 
     T_imp! = SciMLBase.ODEFunction(
-        (Yₜ, u, _, t) -> nothing;
+        (Yₜ, u, _, t) -> (Yₜ .= 0);
```

Comment:
I think I understand why this is needed, but I think we could also easily support the case where `nothing` is returned by ensuring that wherever Y_t ends up to is correctly initialized to 0. (For example, in Rosenbrock, `fU_imp` is initialized to 0, so I think it should be fine to have a function that does `nothing`.)

Reply:
Yeah, agreed. I think it's not a big deal since this is just a CTS test, but it's revealed that we actually have undocumented requirements for users--that the tendencies in some (maybe most?) tendency functions must at least initialize the data to zero, as we currently do not check.

It's currently not easy to support because of what I mentioned [here](https://github.com/CliMA/ClimaTimeSteppers.jl/pull/346#issuecomment-2598779126). I'm going to open a PR in ClimaCore to add support for this.

I think a good future design would be to support passing `Returns(nothing)` instead of an anonymous function, because we could then dispatch on it to literally be a no-op, and we could even check it to see if the update is necessary.

Reply:
> Yeah, agreed. I think it's not a big deal since this is just a CTS test, but it's revealed that we actually have undocumented requirements for users--that the tendencies in some (maybe most?) tendency functions must at least initialize the data to zero, as we currently do not check.

I don't think this should be a requirement: users write functions that evaluate tendendencies, it should be up to the timestepper to ensure that the area of memory used to store these results is properly initialized.


But I agree it's not a big deal, and it's more matter of making clear what's the expectation for the functions that are passed.

Reply:
For the time being, I'm going to split this PR up into two pieces:
 - Fix flaky CI with this change and remove the soft-fail
 - Update `similar` calls to `zero` calls once there is a new version of ClimaCore, so that users cannot fall into this trap

Reply:
There's more design choices with the `similar` vs `zero` business, or having a different default entirely, which we could dispatch on. This change is much less controversial, I think, and fixes the immediate issue of non-deterministic NaNs in the results.

Reply:
I merged this change in #349.